### DISCUSSION TITLE
The directory that WILL NOT die

### DISCUSSION THREAD

**Discussion by FZBunny** [community/community#23854] [New to GitHub]
<p>Hi.  I’m new to both github and git, so this may be eye-rollingly obvious…<br>
When I first created my project in github, I tried to include some binaries in a directory named ‘binaries’.  Didn’t work; when downloaded, they contained only HTML.  Discovered the “Releases” button, and put them there.  So far, so good.  But I can’t get rid of the old ‘binaries’ directory.  Did this:</p>
<pre><code class="lang-auto">rm -rf applepi             # (Just to start from scratch)
git clone ssh://git@github.com/FZBunny/applepi
cd applepi
git rm -r binaries         # dir is now gone...
git push                   #  git replies:  "Everything up-to-date"
cd .. ; rm -rf applepi     # check the result...
git clone ssh://git@github.com/FZBunny/applepi 
cd applepi                 # directory 'binaries'  is back !
</code></pre>
<p>What am I doing wrong?</p>

**Comment by mcaskill** [comment]
<p>Hi,</p>
<p>I believe you need to commit the deleted directory before pushing to the remote repository.</p>
<pre><code class="lang-auto">git rm -r binaries
git commit -m "Removed binaries directory"
git push
</code></pre>
<aside class="onebox stackexchange" data-onebox-src="https://stackoverflow.com/questions/6313126/how-to-remove-a-directory-from-git-repository/27271913#27271913">
  <header class="source">

      <a href="https://stackoverflow.com/questions/6313126/how-to-remove-a-directory-from-git-repository/27271913#27271913" target="_blank" rel="noopener nofollow ugc">stackoverflow.com</a>
  </header>

  <article class="onebox-body">
      <a href="https://stackoverflow.com/users/3540289/suresh-karia" target="_blank" rel="noopener nofollow ugc">
    <img alt="Suresh Karia" src="https://user-images.githubusercontent.com/29353/181079722-6c82a196-675f-413e-bd2c-cd6902c584a4.jpeg" class="thumbnail onebox-avatar" width="128" height="128">
  </a>

<h4>
  <a href="https://stackoverflow.com/questions/6313126/how-to-remove-a-directory-from-git-repository/27271913#27271913" target="_blank" rel="noopener nofollow ugc">How to remove a directory from git repository?</a>
</h4>

<div class="tags">
  <strong>git, github, bitbucket, delete-file</strong>
</div>

<div class="date">
  
  answered by
  <a href="https://stackoverflow.com/users/3540289/suresh-karia" target="_blank" rel="noopener nofollow ugc">
    Suresh Karia
  </a>
  on <a href="https://stackoverflow.com/questions/6313126/how-to-remove-a-directory-from-git-repository/27271913#27271913" target="_blank" rel="noopener nofollow ugc">12:34PM - 03 Dec 14 UTC</a>
</div>

  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

**Comment by FZBunny** [comment]
<p><em>Sigh</em>.  Yes, eye-rollingly obvious.  But it would have taken <em>weeks</em> for it to occur to me.  Thanks, mcaskill!  I owe you a beer.</p>

**Comment by mcaskill** [comment, marked as answer]
<p>We’ve all been there :beers:</p>
